plugins/pencilblue/controllers/actions/admin/sites/edit_site.js

```diff
@@ -0,0 +1,89 @@
+/*
+ Copyright (C) 2015  PencilBlue, LLC
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+module.exports = function EditSiteActionModule(pb) {
+
+    //pb dependencies
+    var util = pb.util;
+
+    /**
+     * Edits a current site
+     */
+    function EditSiteAction(){}
+    util.inherits(EditSiteAction, pb.BaseController);
+
+    EditSiteAction.prototype.render = function(cb)
+    {
+        var self = this;
+        var siteid = this.pathVars.siteid;
+        this.getJSONPostParams(function(err, post) {
```

Comment:
Instead of calling this, in the route declaration include ```request_body: ['application/json']```.  It will instruct the request handler to automatically parse the body for you.  You can specify multiple mime types in the array block.  In your controller you no longer need to call ```getJSONPostParams```.  You can just refer to the parsed body as: ```this.body```.  

It removes the extra level of nesting and extra function call by the controller.